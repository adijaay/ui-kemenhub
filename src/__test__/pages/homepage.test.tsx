import HomePage from "@/pages";
import "@testing-library/jest-dom";
import { render, waitFor } from "@testing-library/react";
import { useRouter } from "next/router";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

jest.mock("@/components/ui/Page", () => {
  return jest.fn().mockImplementation(({ children }) => {
    return <div>{children}</div>;
  });
});

jest.mock("@/utils/firebase", () => ({
  analytics: jest.fn(),
  logEvent: jest.fn(),
}));

describe("HomePage", () => {
  (useRouter as jest.Mock).mockReturnValue({
    query: {},
  });

  it("render home page correctly", async () => {
    const page = render(<HomePage />);
    await waitFor(() => {
      expect(page).toMatchSnapshot();
    });
  });
});
