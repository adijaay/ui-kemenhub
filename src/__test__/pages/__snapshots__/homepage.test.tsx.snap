// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage render home page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
        >
          <p
            class="font-semibold text-[#1f1f1f] text-base"
          >
            Cek Penerima Program
          </p>
          <div
            class="flex flex-col gap-2"
          >
            <p
              class="text-sm font-normal text-[#212121]"
            >
              Nomor Induk Siswa Nasional (NISN)
            </p>
            <div
              class="relative flex items-center"
            >
              <input
                class="h-12 w-full shrink-0 rounded-xl border bg-[#F9FAFB] pl-4 pr-12 text-base font-normal leading-6 text-[#1D2939] outline-none placeholder-ellipsis focus:border-[#1F1F1F] focus:shadow-[0_0_0_2px_#EAECF0] border-[#E1E2EA]"
                data-testid="nisn"
                id="nisn"
                min="0"
                placeholder="Contoh: 112377281"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <p
              class="text-sm font-normal text-[#212121]"
            >
              Nomor Induk Kependudukan (NIK)
            </p>
            <div
              class="relative flex items-center"
            >
              <input
                class="h-12 w-full shrink-0 rounded-xl border bg-[#F9FAFB] pl-4 pr-12 text-base font-normal leading-6 text-[#1D2939] outline-none placeholder-ellipsis focus:border-[#1F1F1F] focus:shadow-[0_0_0_2px_#EAECF0] border-[#E1E2EA]"
                data-testid="nik"
                id="nik"
                min="0"
                placeholder="Contoh: 3530009712399000"
                type="number"
                value=""
              />
            </div>
          </div>
          <button
            class="h-13 rounded-xl bg-[#0C111D] text-base font-semibold text-white"
            data-testid="cekPenerima"
            type="button"
          >
            Cek Penerima
          </button>
        </div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
        >
          <div
            class="flex items-center"
          >
            <div
              class="flex w-full items-start gap-2"
            >
              <img
                alt="info"
                class="h-6 w-6"
                src="/img/target.png"
              />
              <div
                class="flex flex-col gap-2"
              >
                <p
                  class="font-semibold text-[#1f1f1f] text-sm font-bold"
                >
                  Tujuan Program
                </p>
                <p
                  class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px] text-[#667085]"
                >
                  Lihat tujuan dari layanan pendidikan Program Indonesia Pintar.
                </p>
              </div>
            </div>
            <svg
              class="tabler-icon tabler-icon-chevron-right shrink-0"
              fill="none"
              height="20"
              stroke="#D0D5DD"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 6l6 6l-6 6"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
        >
          <div
            class="flex items-center"
          >
            <div
              class="flex w-full items-start gap-2"
            >
              <img
                alt="info"
                class="h-6 w-6"
                src="/img/category.png"
              />
              <div
                class="flex flex-col gap-2"
              >
                <p
                  class="font-semibold text-[#1f1f1f] text-sm font-bold"
                >
                  Kategori Penerima Program
                </p>
                <p
                  class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px] text-[#667085]"
                >
                  Ketahui syarat dan ketentuan penerima Program Indonesia Pintar.
                </p>
              </div>
            </div>
            <svg
              class="tabler-icon tabler-icon-chevron-right shrink-0"
              fill="none"
              height="20"
              stroke="#D0D5DD"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 6l6 6l-6 6"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
        >
          <div
            class="flex items-center"
          >
            <div
              class="flex w-full items-start gap-2"
            >
              <img
                alt="info"
                class="h-6 w-6"
                src="/img/faq.png"
              />
              <div
                class="flex flex-col gap-2"
              >
                <p
                  class="font-semibold text-[#1f1f1f] text-sm font-bold"
                >
                  FAQ (Frequently Ask Question)
                </p>
                <p
                  class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px] text-[#667085]"
                >
                  Daftar pertanyaan terkait Program Indonesia Pintar.
                </p>
              </div>
            </div>
            <svg
              class="tabler-icon tabler-icon-chevron-right shrink-0"
              fill="none"
              height="20"
              stroke="#D0D5DD"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 6l6 6l-6 6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <p
          class="font-semibold text-[#1f1f1f] text-base"
        >
          Cek Penerima Program
        </p>
        <div
          class="flex flex-col gap-2"
        >
          <p
            class="text-sm font-normal text-[#212121]"
          >
            Nomor Induk Siswa Nasional (NISN)
          </p>
          <div
            class="relative flex items-center"
          >
            <input
              class="h-12 w-full shrink-0 rounded-xl border bg-[#F9FAFB] pl-4 pr-12 text-base font-normal leading-6 text-[#1D2939] outline-none placeholder-ellipsis focus:border-[#1F1F1F] focus:shadow-[0_0_0_2px_#EAECF0] border-[#E1E2EA]"
              data-testid="nisn"
              id="nisn"
              min="0"
              placeholder="Contoh: 112377281"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="flex flex-col gap-2"
        >
          <p
            class="text-sm font-normal text-[#212121]"
          >
            Nomor Induk Kependudukan (NIK)
          </p>
          <div
            class="relative flex items-center"
          >
            <input
              class="h-12 w-full shrink-0 rounded-xl border bg-[#F9FAFB] pl-4 pr-12 text-base font-normal leading-6 text-[#1D2939] outline-none placeholder-ellipsis focus:border-[#1F1F1F] focus:shadow-[0_0_0_2px_#EAECF0] border-[#E1E2EA]"
              data-testid="nik"
              id="nik"
              min="0"
              placeholder="Contoh: 3530009712399000"
              type="number"
              value=""
            />
          </div>
        </div>
        <button
          class="h-13 rounded-xl bg-[#0C111D] text-base font-semibold text-white"
          data-testid="cekPenerima"
          type="button"
        >
          Cek Penerima
        </button>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <div
          class="flex items-center"
        >
          <div
            class="flex w-full items-start gap-2"
          >
            <img
              alt="info"
              class="h-6 w-6"
              src="/img/target.png"
            />
            <div
              class="flex flex-col gap-2"
            >
              <p
                class="font-semibold text-[#1f1f1f] text-sm font-bold"
              >
                Tujuan Program
              </p>
              <p
                class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px] text-[#667085]"
              >
                Lihat tujuan dari layanan pendidikan Program Indonesia Pintar.
              </p>
            </div>
          </div>
          <svg
            class="tabler-icon tabler-icon-chevron-right shrink-0"
            fill="none"
            height="20"
            stroke="#D0D5DD"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <div
          class="flex items-center"
        >
          <div
            class="flex w-full items-start gap-2"
          >
            <img
              alt="info"
              class="h-6 w-6"
              src="/img/category.png"
            />
            <div
              class="flex flex-col gap-2"
            >
              <p
                class="font-semibold text-[#1f1f1f] text-sm font-bold"
              >
                Kategori Penerima Program
              </p>
              <p
                class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px] text-[#667085]"
              >
                Ketahui syarat dan ketentuan penerima Program Indonesia Pintar.
              </p>
            </div>
          </div>
          <svg
            class="tabler-icon tabler-icon-chevron-right shrink-0"
            fill="none"
            height="20"
            stroke="#D0D5DD"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <div
          class="flex items-center"
        >
          <div
            class="flex w-full items-start gap-2"
          >
            <img
              alt="info"
              class="h-6 w-6"
              src="/img/faq.png"
            />
            <div
              class="flex flex-col gap-2"
            >
              <p
                class="font-semibold text-[#1f1f1f] text-sm font-bold"
              >
                FAQ (Frequently Ask Question)
              </p>
              <p
                class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px] text-[#667085]"
              >
                Daftar pertanyaan terkait Program Indonesia Pintar.
              </p>
            </div>
          </div>
          <svg
            class="tabler-icon tabler-icon-chevron-right shrink-0"
            fill="none"
            height="20"
            stroke="#D0D5DD"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
